storage2/src/storage.rs

```diff
@@ -29,16 +64,129 @@ impl Storage {
     }
 
     // TODO: this probably can't be 'static long-term
-    pub async fn apply(&'static mut self, state: State) {
-        // TODO: 1. write the index tables and JMT to RocksDB
-        // 2. update the snapshot
-        // TODO: set jmt_version correctly
-        let jmt_version = 0;
-        let snapshot = self.db.snapshot();
-        // Obtain the write-lock for the latest snapshot, and replace it with the new snapshot.
-        let mut guard = self.latest_snapshot.write();
-        *guard = Arc::new(Snapshot::new(snapshot, jmt_version));
-        // Drop the write-lock (this will happen implicitly anyways, but it's good to be explicit).
-        drop(guard);
+    pub async fn apply(&'static mut self, state: State, nct: &Tree) -> Result<()> {
+        // 1. Write the NCT
+        tracing::debug!("serializing NCT");
+        let tct_data = bincode::serialize(nct)?;
+        tracing::debug!(tct_bytes = tct_data.len(), "serialized NCT");
+        gauge!(metrics::NCT_SIZE_BYTES, tct_data.len() as f64);
+
+        let db = self.db;
+
+        let span = Span::current();
+        tokio::task::Builder::new()
+            .name("put_nct")
+            .spawn_blocking(move || {
+                span.in_scope(|| {
+                    let nct_cf = db.cf_handle("nct").expect("nct column family not found");
+                    db.put_cf(nct_cf, "tct", &tct_data)
```

Comment:
Should we call this key "tct" or "nct"? Our only NCT is the TCT, so I'm tempted to say "nct".

Reply:
We were using both in the original code, I started switching over to just nct for the reason you mentioned but missed this :)